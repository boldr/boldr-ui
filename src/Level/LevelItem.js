/* @flow */
import React from 'react';
import classNames from 'classnames';
import { StyleClasses } from '../theme/styleClasses';
import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  isFlexible?: boolean,
  className?: string,
  href?: string,
};
const BASE_ELEMENT = StyleClasses.LEVEL_ITEM;
export function LevelItem({ tag = 'div', ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      'is-flexible': props.isFlexible,
    },
    props.className,
  );

  const { ...HTMLProps } = props;

  return React.createElement(props.href ? 'a' : tag, { ...HTMLProps, className });
}

export default createWrappedComponent(LevelItem);
