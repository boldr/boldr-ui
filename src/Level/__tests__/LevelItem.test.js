import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { LevelItem } from '../LevelItem';

describe('<LevelItem />', () => {
  it('should render a div with .boldr-level__item', () => {
    const component = shallow(<LevelItem>My LevelItem</LevelItem>);
    expect(component.contains(<div className="boldr-level__item">My LevelItem</div>)).toBe(true);
  });

  it('should render a p with .boldr-level__item', () => {
    const component = shallow(<LevelItem tag="p">My LevelItem</LevelItem>);
    expect(component.contains(<p className="boldr-level__item">My LevelItem</p>)).toBe(true);
  });

  it('should render an anchor with .boldr-level__item', () => {
    const component = shallow(<LevelItem href="#">My LevelItem</LevelItem>);
    expect(
      component.contains(
        <a href="#" className="boldr-level__item">
          My LevelItem
        </a>,
      ),
    ).toBe(true);
  });

  it('should render a div with .boldr-level__item and modifiers', () => {
    const component = shallow(<LevelItem isFlexible>My LevelItem</LevelItem>);
    expect(component.hasClass('boldr-level__item')).toBe(true);
    expect(component.hasClass('is-flexible')).toBe(true);
  });

  it('should render a div with .boldr-level__item, modifiers and classNames', () => {
    const component = shallow(
      <LevelItem isFlexible className="custom">
        My LevelItem
      </LevelItem>,
    );
    expect(component.hasClass('boldr-level__item')).toBe(true);
    expect(component.hasClass('is-flexible')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });

  it('should render a div with .boldr-level__item and custom classNames', () => {
    const component = shallow(
      <LevelItem className="custom">
        <span>Any Content</span>
      </LevelItem>,
    );
    expect(component.hasClass('boldr-level__item')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<LevelItem>My LevelItem</LevelItem>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
