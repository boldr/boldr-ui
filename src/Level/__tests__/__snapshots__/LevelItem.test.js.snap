// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelItem /> should match the snapshot 1`] = `
<div
  className="boldrui-level__item"
>
  My LevelItem
</div>
`;
